tests/dashbio_demos/app_speck.py

```diff
@@ -131,22 +266,83 @@ def toggle_zoom_slider_display(scroll_zoom):
 
     @app.callback(
         Output('speck', 'view'),
-        [Input('speck-zoom-slider', 'value')],
-        state=[State('speck', 'view')]
+        [Input('speck-enable-presets', 'values'),
+         Input('speck-atom-radius', 'value'),
+         Input('speck-relative-atom-radius', 'value'),
+         Input('speck-show-hide-bonds', 'values'),
+         Input('speck-bond-scale', 'value'),
+         Input('speck-ao', 'value'),
+         Input('speck-brightness', 'value'),
+         Input('speck-outline', 'value')]
+    )
+    def change_view(
+            presets_enabled,
+            atom_radius,
+            relative_atom_radius,
+            show_bonds,
+            bond_scale,
+            ambient_occlusion,
+            brightness,
+            outline
+    ):
+        return {
+            'atomScale': atom_radius,
+            'relativeAtomScale': relative_atom_radius,
+            'bonds': bool(len(show_bonds) > 0),
+            'bondScale': bond_scale,
+            'ao': ambient_occlusion,
+            'brightness': brightness,
+            'outline': outline
+        }
+
+    @app.callback(
+        Output('speck-store-preset-rendering', 'data'),
+        [Input('speck-preset-rendering-dropdown', 'value')]
     )
-    def zoom_callback(zoom_val, current_view):
-        if current_view is not None:
-            current_view.update(
-                zoom=zoom_val
-            )
-        return current_view
+    def update_rendering_option(render):
```

Comment:
I'm confused about this function... isn't it the identity function?

Reply:
I need this for another callback (`preset_callback`). I need to take into account both inputs from `speck-preset-rendering-dropdown` and `speck-preset-atom-style-dropdown`, but there's no way to see which input triggered the callback. So, I just pass those values into `dcc.Store` components that *do* have a `modified_timestamp` component that allows me to see which dropdown changed more recently. (Pretty convoluted, I know :P)